### DISCUSSION TITLE
Developing WinFSP filesystems with MSYS2/Mingw64 instead of Cygwin?

### DISCUSSION THREAD

**Discussion by lonnietc** [winfsp/winfsp#481] [General]
Hi All,

I have been reading the WinFSP documentation and it talks a lot about using Cygwin for development. I am more partial to MSYS2/MinGW and am trying to determine if anyone has been able to develop with MSYS2 instead of Cygwin as from what I understand about Cygwin, there is a huge DLL shim library that is also needed to actually run anything created in that environment where as MSYS2 compiles things to native.  

I could be wrong on this, but I preferably need to be able to write WinFSP filesystem for Windows but will also eventually be running on Linux, MacOSX, and possible mobile devices as well at some point and was trying to use a single development effort instead of having different code bases.

Of course, there is also Golang instead of C++ but for the time being, I was hoping to just be able to try and compile a couple of Linux FUSE filesystems like MergerFS which is a unionfs and EncSF or similar for an encryption filesystem within a standard environment with minimal patching, if at all possible.

Any thoughts or suggestions would be appreciated greatly.
Thanks, and have a great day.

**Comment by guanzhangrtk** [comment]
Was wondering if anybody else have any feedback on this as I would also like to know whether it is possible to run this in MSYS2 environment. Specifically I would like to be able to create and mount a `ext4` filesystem in MSYS2 and copy some files over. I know that I can use some other third party apps/tools including WSL2 to do this but it would be nice if everything can be self-contained within MSYS2, thanks!
